Add metadata to car

praneelc opened the pull request:
Updated metadata.yaml;

 Source Verification: [https://colab.research.google.com/drive/1ELl_3JTJY_W3Pgr3a6k5rvYpxZ4VGm7z#scrollTo=P6tcV-_JUIvo](https://colab.research.google.com/drive/1ELl_3JTJY_W3Pgr3a6k5rvYpxZ4VGm7z#scrollTo=P6tcV-_JUIvo)

praneelc committed:
Add metadata to car

updated metadata.yaml
 Source Verification: https://colab.research.google.com/drive/1ELl_3JTJY_W3Pgr3a6k5rvYpxZ4VGm7z#scrollTo=P6tcV-_JUIvo

praneelc commented:
Failed check is showing the same error as the dna.yaml PR;
Permission denied to GitHub to access PMLB repo for 'deployment' step in check

trangdata commented:
Hi @praneelc: I noticed one typo in the colab notebook:
```
df_source['doors'] = df_source['doors'].replace(df_source['doors'].unique(), df_pmlb['buying'].unique())
```
On the right, I think it should be `'doors'` not `'buying'`.
After fixing, I think we can remove all the checks and simply do:
```
set(df_pmlb.itertuples(index=False)) == set(df_source.itertuples(index=False))
```
and that should return True. I think some rows are permuted.

trangdata commented on datasets/car/metadata.yaml:
```diff
@@ -1,26 +1,40 @@
+description: >
+  dataset for car evaluation based on six factors;
+  factors considered are buying price, maintenance price, # doors, # people carriable, size of boot, and estimated safety
+source: M. Bohanec, V. Rajkovic. Expert system for decision making. Sistemica 1(1), pp. 145-157, 1990.
```
Can we add the openml link to the source here?

trangdata commented on datasets/car/metadata.yaml:
```diff
@@ -1,26 +1,40 @@
-    code: null # optional, coding information, e.g., Control = 0, Case = 1
-    transform: ~ # optional, any transformation performed on the feature, e.g., log scaled
+    description: buying price of car (vhigh, high, med, low)
+    code: 3 = vhigh, 0 = high, 2 -> med, 1 -> low
```
Can we replace all the `->` with `=`?

trangdata reviewed (commented):
Looks great! I just have a couple comments.

praneelc committed:
Update metadata.yaml

Had some errors in colab, added OpenML source link, and replaced '->' with '='

Using itertuples worked for source check after fixing the typo

trangdata committed:
minor edits
